### ISSUE TITLE
MA0015 is not reported for static Throw methods on Argument(Null)Exception

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#899]
### Version of the Meziantou.Analyzer NuGet package

2.0.235

### Rule Identifier

MAA0015

### Target Framework

.NET 9.0

### C# Language version

4.14.0-3.25412.6 (c167adef)

### Description

**MA0015** is not reported for `Argument(Null)Exception.ThrowIfNull*(...)` while it's correctly reported if you use the constructors of **Argument(Null)Exception**.

### Reproduction Steps

```c#
using System;

internal sealed class Program
{
    private static void Main(string[] args)
    {
        if (Name is null)
        {
            throw new ArgumentNullException(nameof(Name)); // OK
        }

        ArgumentNullException.ThrowIfNull(Name);  // MA0015 is not reported
        ArgumentNullException.ThrowIfNullOrEmpty(Name);  // MA0015 is not reported
        ArgumentNullException.ThrowIfNullOrWhiteSpace(Name); // MA0015 is not reported

        ArgumentException.ThrowIfNullOrWhiteSpace(Name); // MA0015 is not reported
        ArgumentException.ThrowIfNullOrEmpty(Name); // MA0015 is not reported
    }

    public static string Name { get; }
}
```

**Comment by meziantou** [comment]
I agree the rule could be improved to support the new Throw methods. I'll mark it as `Help wanted` as I may not have time to work on it soon.
